### PR TITLE
Adds anomaly detection FAQ items to the Troubleshooting page

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds FAQ items to the Troubleshooting page.

### Preview

[Troubleshooting and FAQ](https://stack-docs_bk_2714.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-ad-troubleshooting.html)

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc [hunk 1]
```diff
@@ -2,14 +2,198 @@
...
+[discrete]
+[[faq-model-accuracy]]
+== How to measure the accuracy of the unsupervised {ml} model?
+
+As model predictions, 95% confidence intervals, and actual values in
+documents in an index (if you enabled model plot) are available, you can use
+standard measures to assess prediction accuracy, interval calibration and so on.
+Aggregations can be used to compute these statistics. This might be useful if 
+you would like to monitor data drifts; however, there are ways to automatically
+adapt to changing data characteristics.
```
Comment (valeriy42) [inline docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc:130]: > 
```suggestion
For each record in a given time series, anomaly detection models provide an anomaly severity score, 95% confidence intervals, and an actual value. This data is stored in an index and can be retrieved using the Get Records API.  With this information, you can use standard measures to assess prediction accuracy, interval calibration, and so on. Elasticsearch aggregations can be used to compute these statistics. 
```

File: docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc [hunk 2]
```diff
@@ -2,14 +2,198 @@
...
+
+The purpose of {anomaly-detect} is to achieve the best ranking of periods where
+an anomaly happened. A useful way to evaluate this is to keep track of real
+incidents and see how well they correlate with the predictions of
+{anomaly-detect}. In practice, it is usually unambiguous qualitatively whether
+{anomaly-detect} is working well.
```
Comment (valeriy42) [inline docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc:136]:
```suggestion
The purpose of {anomaly-detect} is to achieve the best ranking of periods where
an anomaly happened. A practical way to evaluate this is to keep track of real
incidents and see how well they correlate with the predictions of
{anomaly-detect}. 
```

File: docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc [hunk 3]
```diff
@@ -2,14 +2,198 @@
...
+
+
+[discrete]
+[[faq-model-drift]]
+== How to know there is no model drift in unsupervised {ml}?
```
Comment (valeriy42) [inline docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc:141]:
```suggestion
== Can the anomaly detection model experience model drift??
```

File: docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc [hunk 4]
```diff
@@ -2,14 +2,198 @@
...
+* If the predictions are significantly in error over a long period of time, it
+is tested if the time series has undergone a sudden change. Hypothesis testing
+is utilized to check for different types of changes, such as scaling of values,
+shifts in values, and large time shifts in periodic patterns such as daylight
+saving time.
+* Running continuous hypothesis tests on time windows of various lengths to test
+for significant evidence of new or changed periodic patterns, and update the
+model if the null hypothesis of unchanged features is rejected.
+* Accumulating error statistics on calendar days and continually test if
+predictive calendar features must be added to or removed from the model.
```
Comment (valeriy42) [inline docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc:165]:
```suggestion
Elasticsearch's anomaly detection model continuously learns and adapts to changes in the time series. These changes can take the form of slow drifts as well as sudden jumps. Therefore, we take great care to manage the adaptation to changing data characteristics. There is always a fine trade-off between fitting anomalous periods (over-fitting) and not learning new normal behavior. The following are the main approaches Elastic uses to manage this trade-off:

* Learning the optimal decay rate based on measuring the bias in the forecast and the moments of the error distribution
and error distribution moments.
* Allowing continuous small drifts in periodic patterns. This is achieved by
continuously minimizing the mean prediction error over the last iteration with respect to a small bounded time shift.
* If the predictions are significantly wrong over a long period of time, the algorithm
 tests whether the time series has undergone a sudden change. Hypothesis Testing
is used to test for different types of changes, such as scaling of values, shifting of values, and large time shifts in periodic patterns such as daylight saving time.
* Running continuous hypothesis tests on time windows of various lengths to test for significant evidence of new or changed periodic patterns, and update the model if the null hypothesis of unchanged features is rejected. 
* Accumulating error statistics on calendar days and continuously test whether predictive calendar features need to be added or removed from the model.
```

**Review by valeriy42 — COMMENTED** [review]
Hi Istvan! Sorry for the delay. I finally managed to review the page and make some suggestions.

File: docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc [hunk 5]
```diff
@@ -2,14 +2,198 @@
...
+=== Paper
+
+https://www.ijmlc.org/papers/398-LC018.pdf[Veasey, Thomas & Dodson, Stephen. (2014). Anomaly Detection in Application Performance Monitoring Data. International Journal of Machine Learning and Computing. 4. 120-126. 10.7763/IJMLC.2014.V4.398.]
+
+[discrete]
+=== Talks
+
+https://www.elastic.co/elasticon/conf/2018/sf/the-math-behind-elastic-machine-learning[The Math Behind Elastic Machine Learning]
+https://www.elastic.co/elasticon/conf/2017/sf/machine-learning-and-statistical-methods-for-time-series-analysis[Machine Learning and Statistical Methods for Time Series Analysis]
+
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/en/stack/ml/anomaly-detection/ml-ad-troubleshooting.asciidoc:35]: The second link gave me a 404, so I didn't add it to the list.

**Review by valeriy42 — APPROVED** [review]
LGTM. Great work, @szabosteve, on translating my technobabble. This  FAQ will be of great value to the customers. 🚀

**Pull request merged**